service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -41,182 +42,238 @@
 public class RequestStatementMapper {
     private static final Mustache REQUEST_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_request_statement_template.mustache");
 
-    private static final String UBRN = "UBRN: ";
     private static final String UBRN_SYSTEM_URL = "https://fhir.nhs.uk/Id/ubr-number";
-    private static final String PRIORITY = "Priority: ";
-    private static final String SERVICE_REQUESTED = "Service(s) Requested: ";
     private static final String REQUESTER_DEVICE = "Requester Device: ";
-    private static final String REQUESTER_ORG = "Requester Org: ";
     private static final String REQUESTER_PATIENT = "Requester: Patient";
+    private static final String REQUESTER_ORG = "Requester Org: ";
     private static final String REQUESTER_RELATION = "Requester: Relation ";
+    private static final String UBRN = "UBRN: ";
+    private static final String PRIORITY = "Priority: ";
+    private static final String SERVICES_REQUESTED = "Service(s) Requested: ";
     private static final String SPECIALTY = "Specialty: ";
-    private static final String REASON_CODE = "Reason Codes: ";
+    private static final String REASON_CODES = "Reason Codes: ";
+
     private static final String DEFAULT_REASON_CODE_XML = "<code code=\"3457005\" displayName=\"Patient referral\" codeSystem=\"2.16.840.1"
         + ".113883.2.1.3.2.4.15\"/>";
     private static final String NOTE = "Annotation: %s @ %s %s";
     private static final String COMMA = ", ";
 
     private final MessageContext messageContext;
     private final CodeableConceptCdMapper codeableConceptCdMapper;
+    private final ParticipantMapper participantMapper;
 
     public String mapReferralRequestToRequestStatement(ReferralRequest referralRequest, boolean isNested) {
-        var requestStatementTemplateParameters = RequestStatementTemplateParameters.builder()
-            .requestStatementId(messageContext.getIdMapper().getOrNew(ResourceType.ReferralRequest, referralRequest.getId()))
-            .isNested(isNested)
-            .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForReferralRequest(referralRequest))
-            .description(buildDescription(referralRequest))
-            .code(buildCode(referralRequest))
-            .build();
-
-        if (!referralRequest.hasReasonCode()) {
-            requestStatementTemplateParameters.setDefaultReasonCode(DEFAULT_REASON_CODE_XML);
+        return new InnerMapper(referralRequest, isNested).map();
+    }
+
+    @RequiredArgsConstructor
+    private class InnerMapper {
```

Comment:
What is the purpose of the InnerMapper ?

Reply:
A private inner class is a handy way of letting a singleton unsynchronized class do its work with thread-safe state. Helps to simplify code (no need to pass around parameters as much as they can be taken from instance fields instead). I introduced it because of the complexity around creating the text, but even if I change that process as per your other comment I might still prefer to keep the inner class. What do you think?

Reply:
sure, I was not familiar with that approach, we can keep it.